esphome/components/ld2420/ld2420.cpp

```diff
@@ -0,0 +1,506 @@
+      ESP_LOGD(TAG, "LD2420 reply - write register: CMD = %2X %s", CMD_WRITE_REGISTER, result);
+      break;
+    case (CMD_WRITE_ABD_PARAM):
+      ESP_LOGD(TAG, "LD2420 reply - write gate parameter(s): %2X %s", CMD_WRITE_ABD_PARAM, result);
+      break;
+    case (CMD_READ_ABD_PARAM):
+      ESP_LOGD(TAG, "LD2420 reply - read gate parameter(s): %2X %s", CMD_READ_ABD_PARAM, result);
+      data_pos = CMD_ABD_DATA_REPLY_START;
+      for (uint16_t index = 0; index < (CMD_ABD_DATA_REPLY_SIZE *  // NOLINT
+                                        ((buffer[CMD_FRAME_DATA_LENGTH] - 4) / CMD_ABD_DATA_REPLY_SIZE));
+           index += CMD_ABD_DATA_REPLY_SIZE) {
+        memcpy(&cmd_reply_.data[data_element], &buffer[data_pos + index], sizeof(cmd_reply_.data[data_element]));
+        byteswap(cmd_reply_.data[data_element]);
+        ESP_LOGD(TAG, "Data[%2X]: %s", data_element, format_hex_pretty(cmd_reply_.data[data_element]).c_str());
+        data_element++;
+      }
+      break;
+    case (CMD_WRITE_SYS_PARAM):
+      ESP_LOGD(TAG, "LD2420 reply - set system parameter(s): %2X %s", CMD_WRITE_SYS_PARAM, result);
+      break;
+    case (CMD_READ_VERSION):
+      // &buffer[12] - start of version string  buffer[10] sting length
+      memcpy(this->ld2420_firmware_ver_, &buffer[12], buffer[10]);
+      ESP_LOGD(TAG, "LD2420 reply - module firmware version: %7s %s", this->ld2420_firmware_ver_, result);
+      break;
+    default:
+      break;
+  }
+}
+
+int LD2420Component::send_cmd_from_array(CmdFrameT frame) {
+  uint8_t error = 0;
+  uint8_t ack_buffer[64];
+  uint8_t cmd_buffer[64];
+  uint16_t loop_count;
+  cmd_reply_.ack = false;
+  set_cmd_active_(true);
+  uint8_t retry = 3;
+  while (retry) {
+    // TODO setup a dynamic method e.g. millis time count etc. to tune for non ESP32 240Mhz devices
+    // this is ok for now since the module firmware is changing like the weather atm
+    frame.length = 0;
+    loop_count = 1250;
+    uint16_t frame_data_bytes = frame.data_length + 2;  // Always add two bytes for the cmd size
+
+    memcpy(&cmd_buffer[frame.length], &frame.header, sizeof(frame.header));
+    frame.length += sizeof(frame.header);
+
+    memcpy(&cmd_buffer[frame.length], &frame_data_bytes, sizeof(frame.data_length));
+    frame.length += sizeof(frame.data_length);
+
+    memcpy(&cmd_buffer[frame.length], &frame.command, sizeof(frame.command));
+    frame.length += sizeof(frame.command);
+
+    for (uint16_t index = 0; index < frame.data_length; index++) {
+      memcpy(&cmd_buffer[frame.length], &frame.data[index], sizeof(frame.data[index]));
+      frame.length += sizeof(frame.data[index]);
+    }
+
+    memcpy(cmd_buffer + frame.length, &frame.footer, sizeof(frame.footer));
+    frame.length += sizeof(frame.footer);
+    for (uint16_t index = 0; index < frame.length; index++) {
+      this->write_byte(cmd_buffer[index]);
+    }
+
+    ESP_LOGD(TAG, "Tx: %s", format_hex_pretty(cmd_buffer, frame.length).c_str());
```

Comment:
Let's maybe make this verbose or even very-verbose along with those below -- otherwise it really makes the logs unnecessarily noisy.

```suggestion
    ESP_LOGVV(TAG, "Tx: %s", format_hex_pretty(cmd_buffer, frame.length).c_str());
```